README.md

```diff
@@ -11,10 +11,28 @@ $ npm install openwhisk
 
 ## usage
 
+Instantiate openwhisk with specific options:
+
 ```
-var openwhisk = require('openwhisk');
-var options = {apihost: 'openwhisk.ng.bluemix.net', api_key: '...'};
-var ow = openwhisk(options);
+function main(params) {
+	// require the openwhisk npm package
+	var openwhisk = require('openwhisk');
+
+	// read apihost, auth, and namespace from params
+	var apiHost = params.apiHost; # e.g. openwhisk.ng.bluemix.net
+	var namespace = params.namespace;
+	var auth = params.auth; 
+
+	// generate api_key from auth
+	var base64Auth = new Buffer(auth).toString('base64');
+	var apiKey = "Basic " + base64Auth;
```

Comment:
Okay I understand what's happening. 

Without the `return` statement, the Promise from wsk.rules.list fails but is not used the function response and so the openwhisk handler assumes the function returned normally. Once you return the promise, the handler uses that result as the invocation result. 

The wsk.rules.list call fails in both cases but in the first example it's not propagated to the handler.